### PR TITLE
Use flags for gRPC settings

### PR DESCRIPTION
**Pull request by tempoz**
Encapsulate all gRPC flags in the grpc_server package.

---

**Version bump**: TODO

### CODE REVIEW

File: server/libmain/libmain.go
Comment (vadimberezniker) [inline server/libmain/libmain.go:439]: I think this is minor, but this change obscures the fact that `ServeGRPCOverHTTPPort` may or may not serve gRPC over the HTTP server. I think it would be fine to define the flag in this file so that the disable/enable behavior is more obvious.

**Review by vadimberezniker — APPROVED** [review]

File: server/libmain/libmain.go
Comment (tempoz) [reply server/libmain/libmain.go]: IMO, the desire to move this flag out of grpc (which I totally understand) is a result of me having been a little lazy in this CL about encapsulation. To fix this, I moved all of the grpc configuration into a `Register` function and made a grpc over http mux that wraps the env mux. In order to register the grpc_server with just the env, I also put the buildbuddy server and the ssl service in the env (and added registration functions for them) and put the listen address in the env. I also moved the grpc_port and grpcs_port flags into `grpc_server.go`.

File: server/build_event_protocol/build_event_server/build_event_server.go
Comment (vadimberezniker) [inline server/build_event_protocol/build_event_server/build_event_server.go:43]: I think most of these services (apart from the BEP service itself) are out of place in build_event_server.go
They're part of the overall system but are not part of the BEP server.

File: server/util/grpc_server/grpc_server.go
Comment (vadimberezniker) [inline server/util/grpc_server/grpc_server.go:105]: What's the motivation for moving the instantiation of specific services from libmain to this file? 
This file is a utility package (it's under `server/util`) for server support code. The previous location of this code seems more appropriate.

**Review by vadimberezniker — CHANGES_REQUESTED** [review]

File: server/util/grpc_server/grpc_server.go
Comment (tempoz) [reply server/util/grpc_server/grpc_server.go:105]: The motivation is that I would like the grpc Register function to handle actually starting the grpc server, but we can't call `Serve` until all these related services have been registered. I think maybe the right thing to do would be to instantiate them before registering the grpc_server, store them in the env, and then register them?

Comment (tempoz) [reply server/util/grpc_server/grpc_server.go:105]: Oh, except we need one of each for the secure or not secure grpc_server, that's not great. Hmm.

Comment (tempoz) [reply server/util/grpc_server/grpc_server.go:105]: Actually, wait, we use the same buildbuddyServer for both, so presumably it is okay to share instantiations of services between the grpc and grpcs Server

File: server/build_event_protocol/build_event_server/build_event_server.go
Comment (tempoz) [reply server/build_event_protocol/build_event_server/build_event_server.go]: Moved!

File: server/util/grpc_server/grpc_server.go
Comment (tempoz) [reply server/util/grpc_server/grpc_server.go:134]: Put all the registration buts in main now ^-^

**Review by vadimberezniker — APPROVED** [review]

File: server/environment/environment.go
Comment (tempoz) [reply server/environment/environment.go]: Done!

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1778?utm_source=stack-comment).


Job Reference: tNi5q82iqro6xKMPIA1x

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1778?utm_source=stack-comment).


Job Reference: Kb6NfQVs9YSZkiDsKgfx

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1778?utm_source=stack-comment).


Job Reference: ApadsiHESml1IRQRHLJZ

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1778?utm_source=stack-comment).


Job Reference: UWH9zAeATn6HjbQfFiuS

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1778?utm_source=stack-comment).


Job Reference: bO0bzB7R9g7bK8rQ9uPT

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1778?utm_source=stack-comment).


Job Reference: x9vSQMZuwVwdzOiJU1Ph

**Comment by tempoz** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/buildbuddy-io/buildbuddy/1778?utm_source=stack-comment).


Job Reference: xLhUv52COpx6k7AsRAsn

**Pull request merged**
